rclpy/rclpy/executors.py

```diff
@@ -305,93 +291,73 @@ def wait_for_ready_callbacks(self, timeout_sec=None, nodes=None):
+                            yield handler, client, node
 
-            # Process ready entities one node at a time
-            for node in nodes:
-                for tmr in [t for t in node.timers if t.timer_pointer in timers_ready]:
-                    # Check that a timer is ready to workaround rcl issue with cancelled timers
-                    if _rclpy.rclpy_is_timer_ready(tmr.timer_handle):
-                        if tmr.callback_group.can_execute(tmr):
+                    for srv in [s for s in services if wait_set.is_ready(s.service_pointer)]:
+                        if srv.callback_group.can_execute(srv):
                             handler = self._make_handler(
-                                tmr, self._take_timer, self._execute_timer)
+                                srv, self._take_service, self._execute_service)
                             yielded_work = True
-                            yield handler, tmr, node
-
-                for sub in [s for s in node.subscriptions if s.subscription_pointer in subs_ready]:
-                    if sub.callback_group.can_execute(sub):
-                        handler = self._make_handler(
-                            sub, self._take_subscription, self._execute_subscription)
-                        yielded_work = True
-                        yield handler, sub, node
-
-                for gc in [g for g in node.guards if g._executor_triggered]:
-                    if gc.callback_group.can_execute(gc):
-                        handler = self._make_handler(
-                            gc, self._take_guard_condition, self._execute_guard_condition)
-                        yielded_work = True
-                        yield handler, gc, node
-
-                for client in [c for c in node.clients if c.client_pointer in clients_ready]:
-                    if client.callback_group.can_execute(client):
-                        handler = self._make_handler(
-                            client, self._take_client, self._execute_client)
-                        yielded_work = True
-                        yield handler, client, node
-
-                for srv in [s for s in node.services if s.service_pointer in services_ready]:
-                    if srv.callback_group.can_execute(srv):
-                        handler = self._make_handler(
-                            srv, self._take_service, self._execute_service)
-                        yielded_work = True
-                        yield handler, srv, node
-
-            # Check timeout timer
-            if (timeout_nsec == 0 or
-                    (timeout_timer is not None and timeout_timer.timer_pointer in timers_ready)):
-                break
+                            yield handler, srv, node
+
+                # Check timeout timer
+                if (timeout_nsec == 0 or
+                        (timeout_timer is not None and wait_set.is_ready(
+                            timeout_timer.timer_pointer))):
```

Comment:
I would suggest (without having run linters on the following snippet :wink:):

```
if (
    timeout_nsec == 0 or
    (timeout_timer and wait_set.is_ready(timeout_timer.timer_pointer))
):
```

Reply:
Linter is happy 
ded618f

Reply:
Getting rid of `is not None` would make the line shorter but it is ok as-is.